lisdf/parsing/urdf_j.py

```diff
@@ -0,0 +1,312 @@
+        )
+
+        limit = None
+        limit_node = node.pop("limit", return_type="node", default=None)
+        if limit_node is not None:
+            limit = C.JointLimit(
+                lower=safe_float(limit_node.attributes.pop("lower", None)),
+                upper=safe_float(limit_node.attributes.pop("upper", None)),
+                effort=safe_float(limit_node.attributes.pop("effort", None)),
+                velocity=safe_float(limit_node.attributes.pop("velocity", None)),
+            )
+        dynamics = None
+        dynamics_node = node.pop("dynamics", return_type="node", default=None)
+        if dynamics_node is not None:
+            dynamics = C.JointDynamics(
+                damping=float(dynamics_node.attributes.pop("damping", 0)),
+                friction=float(dynamics_node.attributes.pop("friction", 0)),
+            )
+        calibration = None
+        calibration_node = node.pop("calibration", return_type="node", default=None)
+        if calibration_node is not None:
+            calibration = C.JointCalibration(
+                rising=float(calibration_node.attributes.pop("rising", 0)),
+                falling=float(calibration_node.attributes.pop("falling", 0)),
+            )
+        control_info = None
+        control_node = node.pop("safety_controller", return_type="node", default=None)
+        if control_node is not None:
+            control_info = C.JointControlInfo(
+                lower=safe_float(control_node.attributes.pop("soft_lower_limit", None)),
+                upper=safe_float(control_node.attributes.pop("soft_upper_limit", None)),
+                position=safe_float(control_node.attributes.pop("k_position", None)),
+                velocity=safe_float(control_node.attributes.pop("k_velocity", None)),
+            )
+        mimic_info = None
+        mimic_node = node.pop("mimic", return_type="node", default=None)
+        if mimic_node is not None:
+            mimic_info = C.JointMimic(
+                joint=mimic_node.attributes.pop("joint"),
+                multiplier=float(mimic_node.attributes.pop("multiplier", 1)),
+                offset=float(mimic_node.attributes.pop("offset", 0)),
+            )
+
+        return node.set_data(
+            C.Joint(
+                name=name,
+                pose=pose,
+                parent=parent,
+                child=child,
+                joint_info=C.JointInfo.from_type(
+                    type,
+                    axis=axis,
+                    limit=limit,
+                    dynamics=dynamics,
+                    calibration=calibration,
+                    mimic=mimic_info,
+                ),
+                control_info=control_info,
+            )
+        )
+
+    def transmission_init(self, node):
+        return "skip"  # TODO: implement transmission.
+
+    def gazebo_init(self, node):
+        return "skip"
+
+    def verbose_init(self, node):
+        return "skip"
```

Comment:
would it be useful to just raise an exception here so that user's don't accidentally believe these are currently implemented?